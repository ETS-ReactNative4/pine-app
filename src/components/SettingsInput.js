import React, { Component } from 'react';
import { StyleSheet, View, TextInput } from 'react-native';
import PropTypes from 'prop-types';

const styles = StyleSheet.create({
  container: {
    paddingTop: 12,
    paddingBottom: 12,
    paddingRight: 15,
    paddingLeft: 15
  },
  input: {
    fontFamily: 'System',
    fontWeight: '400',
    fontSize: 17,
    color: '#000000'
  }
});

export default class SettingsInput extends Component {
  render() {
    return (
      <View style={styles.container}>
        <TextInput
          style={styles.input}
          value={this.props.value}
          placeholder={this.props.placeholder}
          placeholderTextColor='#8E8E93'
          selectionColor='#426BF2'
          clearButtonMode='always'
          autoFocus={true}
          keyboardType={this.props.keyboardType}
          returnKeyType='done'
          onChangeText={this.props.onChangeText}
          onSubmitEditing={this.props.onSubmitEditing}
          maxLength={this.props.maxLength}
        />
      </View>
    );
  }
}

SettingsInput.propTypes = {
  value: PropTypes.string,
  placeholder: PropTypes.string,
  keyboardType: PropTypes.string,
  onChangeText: PropTypes.func,
  onSubmitEditing: PropTypes.func,
  maxLength: PropTypes.number
};
