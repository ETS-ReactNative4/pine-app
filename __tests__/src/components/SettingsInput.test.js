import React from 'react';
import renderer from 'react-test-renderer';
import SettingsInput from '../../../src/components/SettingsInput';

describe('SettingsInput', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <SettingsInput
        value='5f2d5a31-66cc-438f-9136-085e1c8ceb2b'
        placeholder='f69225ae-ee9b-4e45-9688-4a74ae3b82e9'
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
