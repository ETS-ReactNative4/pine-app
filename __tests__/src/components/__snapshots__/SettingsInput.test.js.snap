// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsInput renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "paddingBottom": 12,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 12,
      },
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    autoFocus={true}
    clearButtonMode="always"
    editable={true}
    placeholder="f69225ae-ee9b-4e45-9688-4a74ae3b82e9"
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Array [
        Object {
          "fontFamily": "System",
          "fontSize": 17,
          "fontWeight": "400",
        },
        Object {
          "color": "#000000",
        },
      ]
    }
    underlineColorAndroid="transparent"
    value="5f2d5a31-66cc-438f-9136-085e1c8ceb2b"
  />
</View>
`;
